src/registry/radio/preview.tsx

```diff
@@ -22,36 +26,47 @@ const Preview: React.FC<ComponentPreviewProps<RadioComponentSchema>> = ({compone
   const intl = useIntl();
   const {key, label, description, tooltip, validate} = component;
   const {required = false} = validate || {};
+  const {getReferentielijstenTabelItems} = useContext(BuilderContext);
 
-  let options: Option[] = [];
-  if (checkIsManualOptions(component)) {
-    options = component?.values || [];
-  } else if (checkIsReferentielijstenOptions(component)) {
-    options = [
-      {
-        value: 'option1',
-        label: intl.formatMessage({
-          description: 'Radio dummy option1 from referentielijsten',
-          defaultMessage: 'Option from referentielijsten: option1',
-        }),
-      },
-    ];
-  } else if (checkIsVariableOptions(component)) {
-    options = [
-      {
-        value: 'itemsExpression',
-        label: intl.formatMessage(
-          {
-            description: 'Radio dummy option for itemsExpression',
-            defaultMessage: 'Options from expression: <code>{expression}</code>',
-          },
-          {
-            expression: JSON.stringify(component.openForms.itemsExpression),
-            code: chunks => <code>{chunks}</code>,
-          }
-        ),
-      },
-    ];
+  const {
+    value: options = [],
+    loading,
+    error,
+  } = useAsync(async () => {
+    if (checkIsManualOptions(component)) {
+      return component?.values || [];
+    }
+    if (checkIsVariableOptions(component)) {
+      return [
+        {
+          value: 'itemsExpression',
+          label: intl.formatMessage(
+            {
+              description: 'Selectboxes dummy option for itemsExpression',
+              defaultMessage: 'Options from expression: <code>{expression}</code>',
+            },
+            {
+              expression: JSON.stringify(component.openForms.itemsExpression),
+              code: chunks => <code>{chunks}</code>,
+            }
+          ),
+        },
+      ];
+    }
+
+    if (checkIsReferentielijstenOptions(component)) {
+      const response: ReferentielijstenTabelItem[] = await getReferentielijstenTabelItems(
```

Comment:
```suggestion
      const items: ReferentielijstenTabelItem[] = await getReferentielijstenTabelItems(
```
`response` implies it's an HTTP/API response, but the return type does not agree with that.